### ISSUE TITLE
AzCopy v10 "panic: send on closed channel" on 403 Response

### ISSUE THREAD

**Issue by tommywilkinson** [Azure/azure-storage-azcopy#2703]
### Which version of the AzCopy was used? 
10.25

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

### What command did you run?
"copy <local path> <destination> --log-level=INFO --recursive"
(redacted exact details, since it's not terribly relevant to the repro)

### What problem was encountered?
User account had owner permissions on storage account but not Storage Blob Data Contributor. Full error message from running azcopy locally:

```
INFO: Authentication failed, it is either not correct, or expired, or does not have the correct permission PUT https://<storageaccount>.blob.core.windows.net/1js/pr/3045226/wdjs/wac/we-c/debug/suiteux-shell/js/suiteux.shell.npm_mecontrol.rtl.js
--------------------------------------------------------------------------------
RESPONSE 403: 403 This request is not authorized to perform this operation using this permission.
ERROR CODE: AuthorizationPermissionMismatch
--------------------------------------------------------------------------------
﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>AuthorizationPermissionMismatch</Code><Message>This request is not authorized to perform this operation using this permission.
RequestId:3cd177f3-c01e-0050-18a5-b6a35a000000
Time:2024-06-04T17:34:48.4836791Z</Message></Error>
--------------------------------------------------------------------------------

panic: send on closed channel

goroutine 140 [running]:
github.com/Azure/azure-storage-azcopy/v10/ste.(*jobMgr).SendXferDoneMsg(0xc037b4fe08?, {{0xc000ef3b00, 0x7a}, {0xc000ef3b80, 0x80}, 0x0, 0xfffffffa, 0x12385, 0x0})
        D:/a/1/s/ste/jobStatusManager.go:72 +0x4e
github.com/Azure/azure-storage-azcopy/v10/ste.(*jobPartMgr).SendXferDoneMsg(0x180a2c3?, {{0xc000ef3b00, 0x7a}, {0xc000ef3b80, 0x80}, 0x0, 0xfffffffa, 0x12385, 0x0})
        D:/a/1/s/ste/mgr-JobPartMgr.go:694 +0x6e
github.com/Azure/azure-storage-azcopy/v10/ste.(*jobPartTransferMgr).ReportTransferDone(0xc000f08230)
        D:/a/1/s/ste/mgr-JobPartTransferMgr.go:992 +0x16c
github.com/Azure/azure-storage-azcopy/v10/ste.(*jobMgr).transferProcessor.func1({0x1d7ba20, 0xc000f08230})
        D:/a/1/s/ste/mgr-JobMgr.go:1089 +0xe2
github.com/Azure/azure-storage-azcopy/v10/ste.(*jobMgr).transferProcessor(0xc00039a388, 0x6acaaa856acaaa85?)
        D:/a/1/s/ste/mgr-JobMgr.go:1105 +0xb2
created by github.com/Azure/azure-storage-azcopy/v10/ste.NewJobMgr in goroutine 1
        D:/a/1/s/ste/mgr-JobMgr.go:200 +0xbf1
panic: send on closed channel
```

Not a big issue, but seems like a relatively expected case to be causing a panic. 

### How can we reproduce the problem in the simplest way?
Run azcopy in some manner to get a 403 response. 

### Have you found a mitigation/solution?
Yes. It's just a noisy failure. It prevents ending statistics from being printed properly, but otherwise is not very impactful.

**Comment by seanmcc-msft** [comment]
@tommywilkinson thanks for reporting this, we have added this item to our backlog to be fixed in a future release.

**Comment by seanmcc-msft** [comment]
@tommywilkinson what type of authentication are you using in your reproduction?  AAD?

**Comment by wonwuakpa-msft** [comment]
Hi @tommywilkinson, we have a [fix](https://github.com/Azure/azure-storage-azcopy/pull/2824) for this in our last release. Please update to version 10.27.1 after it is released later today.

**Comment by gapra-msft** [comment]
Closing as this is resolved by 10.27.1
